src/DucksNet.API/Controllers/AppointmentsController.cs

```diff
@@ -10,9 +10,16 @@ namespace DucksNet.API.Controllers;
 public class AppointmentsController : ControllerBase
 {
     private readonly IRepository<Appointment> _appointmentsRepository;
-    public AppointmentsController(IRepository<Appointment> repository)
+    private readonly AppointmentScheduleService _appointmentScheduleService;
+    private readonly IRepository<Office> _officeRepository;
+    private readonly IRepository<Pet> _animalRepository;
+
+    public AppointmentsController(IRepository<Appointment> repository, IRepository<Office> officeRepository, IRepository<Pet> animalRepository)
     {
         _appointmentsRepository = repository;
+        _officeRepository = officeRepository;
+        _animalRepository = animalRepository;
+        _appointmentScheduleService = new AppointmentScheduleService(repository, officeRepository, animalRepository);
```

Comment:
need some explanation here

Reply:
The service handles scheduling irrespective of repository implementation. So we inject them here trough constructor.